test/sql/append.sql

```diff
@@ -0,0 +1,74 @@
+\ir include/create_single_db.sql
+
+-- create a now() function for repeatable testing that always returns
+-- the same timestamp. It needs to be marked STABLE
+CREATE OR REPLACE FUNCTION now_s()
+RETURNS timestamp LANGUAGE PLPGSQL STABLE AS
+$BODY$
+BEGIN
+     RETURN '2017-08-22T10:00:00'::timestamp;
+END;
+$BODY$;
+
+CREATE OR REPLACE FUNCTION now_i()
+RETURNS timestamp LANGUAGE PLPGSQL IMMUTABLE AS
+$BODY$
+BEGIN
+     RETURN '2017-08-22T10:00:00'::timestamp;
+END;
+$BODY$;
+
```

Comment:
Unfortunately, I haven't found a function that checks for only stable functions in expressions, so in case of only volatile functions in the expression, we might apply the optimization needlessly, although I don't think it would do anything wrong.